engine/schema/src/main/resources/META-INF/db/schema-41310to41400.sql

```diff
@@ -379,3 +379,81 @@ CREATE TABLE IF NOT EXISTS `cloud`.`kubernetes_cluster_details` (
     PRIMARY KEY(`id`),
     CONSTRAINT `fk_kubernetes_cluster_details__cluster_id` FOREIGN KEY `fk_kubernetes_cluster_details__cluster_id`(`cluster_id`) REFERENCES `kubernetes_cluster`(`id`) ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+ALTER TABLE `cloud`.`roles` ADD COLUMN `is_default` tinyint(1) NOT NULL DEFAULT '0' COMMENT 'is this a default role';
+UPDATE `cloud`.`roles` SET `is_default` = 1 WHERE id IN (1, 2, 3, 4);
+
+-- Updated Default CloudStack roles with read-only and support admin and user roles
+INSERT INTO `cloud`.`roles` (`uuid`, `name`, `role_type`, `description`, `is_default`) VALUES (UUID(), 'Read-Only Admin', 'Admin', 'Default read-only admin role', 1) ON DUPLICATE KEY UPDATE name=name;
```

Comment:
Why not change the name to reflect these are default or `built-in` roles, should we be renaming any existing roles matching this? It could lead to side-effects, I would've preferred failing an upgrade instead of silent override.

Reply:
this is in sync with existing default roles with id (1,2,3,4). Shall I prepend all these role names with "Default" ?